pkg/api/validation/validation.go

```diff
@@ -735,6 +743,32 @@ func validateConfigMapVolumeSource(configMapSource *api.ConfigMapVolumeSource, f
 	return allErrs
 }
 
+func validateHugePagesVolumeSource(hugePages *api.HugePagesVolumeSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// TODO(pprokop): add support for other PageSizes
+	if hugePages.PageSize != "2M" {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("pageSize"), hugePages.PageSize, "Not supported page size"))
+	}
+	maxSize, err := resource.ParseQuantity(hugePages.MaxSize)
```

Comment:
Kernel itself rounds down size to  HPAGE_SIZE boundary, so if you specify 5M it will be 2 pages.